File: for_my_blog/parsing_eol/parse_csv.rb
#!/usr/bin/env ruby

require 'csv'

parsings = []

raise "Missing file(s) to parse" unless ARGV.length > 0

ARGV.each {|file|
  puts "About to parse: #{file}"
  parsed = CSV.read(file)
  parsings.push(parsed)
  parsed.each {|line|
    puts "\tNEW LINE :"
    line.each {|token| puts "\t\t '#{token}'" }
  }
  puts "---end---"
}

# si on a passé plusieurs fichiers, indiquer s'ils ont été parsés différemment :
reference = parsings[0]
parsings.each{ |parsing|
  if reference != parsing
    puts "WARNING : input files were parsed differently !"
  end
}

File: for_my_blog/parsing_eol/parse_csv.py
#!/usr/bin/env python3

import csv
import sys
from functools import reduce

if len(sys.argv) <= 1:
    raise RuntimeError("Missing file(s) to parse")

filenames = sys.argv[1:]
parsings = []
for filename in filenames:
    print("About to parse: {}".format(filename))
    with open(filename, "r") as f:
        reader = csv.reader(f)
        parsing = []

        for tokens in reader:
            print("\tNEW LINE :")
            this_line = []
            for token in tokens:
                print("\t\t '{}'".format(token))
                this_line.append(token)
            parsing.append(this_line)
        print("---end---")
    parsings.append(parsing)

all_files_were_parsed_identically = reduce(lambda a, b: a == b, parsings)
if not all_files_were_parsed_identically:
    print("WARNING : input files were parsed differently !")
